nodejs/week3/practices/exercise-template/src/backend/api/concerts.js

```diff
@@ -0,0 +1,169 @@
+const express = require("express");
+const { request } = require("../app");
+const router = express.Router();
+const knex = require("../database");
+
+
+router.get("/", async (request, response) => {
+  const concerts = await knex("concerts");
+ 
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+          if(request.query.hasOwnProperty("id")){
+            
+            const requestId=Number(request.query.id);
+            const concertFinder = concerts.some(concert=> concert.id ===requestId);
+            const requestedConcert = await concerts.filter(concert=> concert.id ===requestId);
```

Comment:
I don't think you need both ``concertFinder`` and ``requestedConcert`` here. The latter is enough for this excercise.